src/actions/arcActions.ts

```diff
@@ -674,19 +665,37 @@ export function createProposal(daoAvatarAddress: string, title: string, descript
       // Go back to home page while action create proposal operation gets carried out
       dispatch(push("/dao/" + daoAvatarAddress));
 
-      await contributionRewardInstance.proposeContributionReward({
-        title,
-        avatar: daoAvatarAddress,
-        beneficiaryAddress,
-        description,
-        ethReward: Util.toWei(ethReward),
-        externalToken: dao.externalTokenAddress ? dao.externalTokenAddress : "",
-        externalTokenReward: dao.externalTokenAddress && externalTokenReward ? Util.toWei(externalTokenReward) : 0,
-        nativeTokenReward: Util.toWei(nativeTokenReward),
-        numberOfPeriods: 1,
-        periodLength: 0,
-        reputationChange: Util.toWei(reputationReward),
-      } as any);
+      const dao = getState().arc.daos[daoAvatarAddress];
+      const contributionRewardInstance = await Arc.ContributionRewardFactory.deployed();
+      const wallet = await getWallet()
+      const contributionReward = new ethers.Contract(contributionRewardInstance.address, contributionRewardArtifacts.abi, wallet)
+
+      if (!beneficiaryAddress.startsWith("0x")) { beneficiaryAddress = "0x" + beneficiaryAddress; }
+      beneficiaryAddress = beneficiaryAddress.toLowerCase();
+
+      const hash = ipfs.hashToHex(ipfsHash)
+
+      const externalTokenAddress = dao.externalTokenAddress ? dao.externalTokenAddress : "0x0000000000000000000000000000000000000000"
+
+      const arg = [
+        daoAvatarAddress,
+        hash,
+        Util.toWei(reputationReward),
+        [
+          Util.toWei(nativeTokenReward),
+          Util.toWei(ethReward),
+          Util.toWei(externalTokenReward),
+          "0",
+          "1"
+        ],
+        externalTokenAddress,
+        beneficiaryAddress
+      ]
+
+      const gasLimit = await contributionReward.estimate.proposeContributionReward(arg[0], arg[1], arg[2], arg[3], arg[4], arg[5])
+      const txReceipt = await contributionReward.proposeContributionReward(arg[0], arg[1], arg[2], arg[3], arg[4], arg[5], { gasLimit })
+
+      console.log(JSON.stringify(txReceipt,null,2))
```

Comment:
remove console.log. this should have been caught by the linter

Reply:
I think logging this is useful, is there another logging framework I should use other than console.log?

Reply:
arc.js has a [logging subsystem](https://daostack.github.io/arc.js/api/classes/LoggingService/)